src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -186,19 +200,15 @@ internal unsafe SocketError DoOperationAccept(Socket socket, SafeCloseSocket han
                     out int bytesTransferred,
                     overlapped);
 
-                socketError = ProcessIOCPResult(success, bytesTransferred);
-                return socketError;
+                return ProcessIOCPResult(success, bytesTransferred, overlapped);
```

Comment:
I realize this is the current pattern, but...

It seems slightly weird that we call ProcessIOCPResult inside the try/catch.  ProcessIOCPResult should never throw, so it's slightly cleaner to call it outside the try.  But if it did throw, seems like we'd free the NativeOverlapped twice.

Maybe not worth worrying about for this PR, just something to consider...